org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/TargetNamespace_1CodeAction.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2020 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Red Hat Inc. - initial API and implementation
+ */
+package org.eclipse.lemminx.extensions.contentmodel.participants.codeactions;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import org.eclipse.lemminx.commons.CodeActionFactory;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.services.extensions.ICodeActionParticipant;
+import org.eclipse.lemminx.services.extensions.IComponentProvider;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Range;
+
+/**
+ * CodeAction to replace an incorrect namespace in an .xml document.
+ * 
+ * Changes the value of the xmlns attribute of the root element of the .xml
+ * document to the declared namespace of the referenced .xsd document.
+ */
+public class TargetNamespace_1CodeAction implements ICodeActionParticipant {
+
+	private static final Pattern NAMESPACE_EXTRACTOR = Pattern.compile("'([^']+)'\\.");
+
+	@Override
+	public void doCodeAction(Diagnostic diagnostic, Range range, DOMDocument document, List<CodeAction> codeActions,
+			SharedSettings sharedSettings, IComponentProvider componentProvider) {
+
+		String namespace = extractNamespace(diagnostic.getMessage());
```

Comment:
if namespace is empty/null, you should return early

Reply:
This does make more sense as a fail-safe behaviour; I'll implement it.

It also made me think of this test case. The expected behaviour would be hard to implement, since `null` is a valid namespace:

![replace-namespace-with-null](https://user-images.githubusercontent.com/22376627/83895784-2e411d00-a721-11ea-85bd-6019b9e13402.gif)

Reply:
With this usecase, You have found a new bug with error range with EmptyTargetNamespace (see orange warning at the end of xml-model). You can create a bug for this issue.

I think it's a bug of Xerces which should generate a message with 'empty namespace' instead of null. You can create a second bug for this issue.

Reply:
I created GitHub issues for these issues.